File: tests/fixtures/portable-map-evaluation/sources/mixed-monorepo/apps/dispatch-console/src/application/contracts.ts
import type { DispatchEvent } from "../types.ts";

export function parseDispatchEvent(value: unknown): DispatchEvent {
  if (!value || typeof value !== "object") throw new Error("dispatch event must be an object");
  const candidate = value as Record<string, unknown>;
  const fields = ["dispatchId", "boxId", "depotCode", "routeDate"];
  if (candidate.eventType !== "dispatch.assigned" || candidate.status !== "ASSIGNED") {
    throw new Error("only assigned dispatch events are accepted");
  }
  for (const field of fields) {
    if (typeof candidate[field] !== "string" || candidate[field] === "") throw new Error(`missing ${field}`);
  }
  return candidate as unknown as DispatchEvent;
}

File: tests/fixtures/portable-map-evaluation/sources/mixed-monorepo/workers/route-planner/src/routeplanner/contracts.py
import json
import sys

from .errors import PlanningError


def validate_dispatch_event(value: object) -> tuple[str, str]:
    if not isinstance(value, dict):
        raise PlanningError("dispatch event must be an object")
    required = ("eventType", "dispatchId", "boxId", "depotCode", "routeDate", "status")
    if any(not isinstance(value.get(field), str) or not value[field] for field in required):
        raise PlanningError("dispatch event fields are required")
    if value["eventType"] != "dispatch.assigned" or value["status"] != "ASSIGNED":
        raise PlanningError("dispatch event is not an assigned event")
    return value["eventType"], value["status"]


def main() -> None:
    if sys.argv[1:] != ["--probe"]:
        raise SystemExit("usage: python -m routeplanner.contracts --probe")
    event_type, status = validate_dispatch_event(json.load(sys.stdin))
    print(f"{event_type}:{status}")


if __name__ == "__main__":
    main()
